packages/adapters/src/proces-sequence-adapter.ts

```diff
@@ -0,0 +1,114 @@
+import { getLogger } from "@scramjet/logger";
+import {
+    Logger,
+    ProcessRunnerConfig,
+    ISequenceAdapter,
+    ISequenceInfo,
+} from "@scramjet/types";
+import { Readable } from "stream";
+import { createReadStream, createWriteStream, existsSync } from "fs";
+import { mkdir, readdir, rm } from "fs/promises";
+import * as path from 'path'
+import { exec } from "child_process";
+import { isDefined, readStreamedJSON } from "@scramjet/utility";
+import { isValidSequencePackageJSON } from "@scramjet/types";
+import { SequenceInfo } from "./sequence-info";
+
+const HOME_DIR = require('os').homedir();
+const SEQUENCES_DIR = path.join(HOME_DIR, '.scramjet_sequences')
+
+async function getRunnerConfigForStoredSequence(id: string): Promise<ProcessRunnerConfig> {
+    const packageJsonPath = path.join(SEQUENCES_DIR, id, 'package.json')
+    const packageJson = await readStreamedJSON(createReadStream(packageJsonPath))
+
+    if(!isValidSequencePackageJSON(packageJson)) {
+        throw new Error('Invalid Scramjet sequence package.json')
+    }
+
+    return {
+        type: 'process',
+        sequencePath: packageJson.main,
+        version: packageJson.version ?? '',
+        name: packageJson.name ?? '',
+        id   
+    }
+}
+
+export function getSequenceDir(id: string): string {
+    return path.join(SEQUENCES_DIR, id)
+}
+
+class ProcessSequenceAdapter implements ISequenceAdapter {
+    private logger: Logger;
+    private computedInfo: ISequenceInfo | null = null
+
+    constructor(info?: ISequenceInfo) {
+        if(info && info.getConfig().type !== 'process') {
+            throw new Error('Invalid info config for ProcessSequenceAdapter')
+        }
+
+        this.computedInfo = info ?? null;
+        this.logger = getLogger(this);
+    }
+
+    public get info(): ISequenceInfo {
+        if(!this.computedInfo) {
+            throw new Error('Sequence not identified yet');
+        }
+
+        return this.computedInfo
+    }
+
+    async init(): Promise<void> {
+        if(!existsSync(SEQUENCES_DIR)) {
+            await mkdir(SEQUENCES_DIR)
+        }
+    }
+    
+    async list(): Promise<ProcessSequenceAdapter[]> {
+        const storedSequencesIds = await readdir(SEQUENCES_DIR)
+        const sequencesConfigs = await Promise.all(
+            storedSequencesIds
+                .map(getRunnerConfigForStoredSequence)
+                .map((configPromised) => configPromised.catch(() => null))
+        )
+
+        this.logger.debug('Listed stored sequences', sequencesConfigs)
+
+
```

Comment:
double empty line

Reply:
oh i see after reinstalling vscode server i forgot to reinstall eslint extension. on it :wink: